### PR TITLE
PMM-12872 Fallback to PGSS in case of not available PGSM.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-12872

Link to the Feature Build: SUBMODULES-0

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

File: managed/services/management/postgresql.go [hunk 1]
```diff
@@ -119,6 +123,13 @@ func (s *PostgreSQLService) Add(ctx context.Context, req *managementpb.AddPostgr
 			if err = s.sib.GetInfoFromService(ctx, tx.Querier, service, row); err != nil {
 				return err
 			}
+
+			// In case of not available PGSM extension it is switch to PGSS.
+			if req.QanPostgresqlPgstatmonitorAgent && row.PostgreSQLOptions != nil && row.PostgreSQLOptions.PGSMVersion == "" {
+				s.l.Warningln("Extension pg_stat_monitor is not available. Switching to pg_stats_statements.")
```
Comment (BupycHuk) [inline managed/services/management/postgresql.go:129]: what do you think, should we show warning in pmm-admin?

File: api/agentpb/agent.proto [hunk 2]
```diff
@@ -474,6 +474,8 @@ message ServiceInfoResponse {
   string version = 3;
   // A list of PostgreSQL databases.
   repeated string database_list = 4;
+  // A version of PGSM, empty if PGSM is not available.
```
Comment (ademidoff) [inline api/agentpb/agent.proto:477]:
```suggestion
  // A version of pg_stat_monitor, empty if unavailable.
```

File: managed/services/management/postgresql.go [hunk 3]
```diff
@@ -46,6 +48,7 @@ func NewPostgreSQLService(db *reform.DB, state agentsStateUpdater, cc connection
 		state: state,
 		cc:    cc,
 		sib:   sib,
+		l:     logrus.WithField("component", "postgesql"),
```
Comment (ademidoff) [inline managed/services/management/postgresql.go:51]:
```suggestion
		l:     logrus.WithField("component", "postgresql"),
```

File: managed/services/management/postgresql.go (hunk 1 above)
Comment (ademidoff) [reply managed/services/management/postgresql.go:129]: The ticket says `pg_stats_statements` should be the default, so we probably shouldn't even mention that `pg_stat_monitor` is unavailable - it's not something we event want to distract the user with.

Comment (BupycHuk) [reply managed/services/management/postgresql.go:129]: Actually behaviour implemented by @JiriCtvrtka in this PR is what we wanted initially in [PMM-10974](https://perconadev.atlassian.net/browse/PMM-10974)

Comment (JiriCtvrtka) [reply managed/services/management/postgresql.go:129]: Right now I am updating ticket to be aligned with old one PMM-10974. 

About warning I would prefer warning (maybe info) in PMM server to dont distract user. Users with PGSM will have it and PGSS users not even notice until they take a look into server logs.

Comment (ademidoff) [reply managed/services/management/postgresql.go:129]: Well, I'm looking at the commit statements - it says `PMM-12872` eveyrwhere, not `PMM-10974`. Not sure really what to look at :)

Comment (JiriCtvrtka) [reply managed/services/management/postgresql.go:129]: Ticket is updated. So please take a look if it is aligned now.

File: managed/services/management/postgresql.go [hunk 4]
```diff
@@ -119,6 +123,13 @@ func (s *PostgreSQLService) Add(ctx context.Context, req *managementpb.AddPostgr
 			if err = s.sib.GetInfoFromService(ctx, tx.Querier, service, row); err != nil {
 				return err
 			}
+
+			// In case of not available PGSM extension it is switch to PGSS.
+			if req.QanPostgresqlPgstatmonitorAgent && row.PostgreSQLOptions != nil && row.PostgreSQLOptions.PGSMVersion == "" {
```
[the remark this answers is not in the document]
Comment (JiriCtvrtka) [reply managed/services/management/postgresql.go]: Yeah not necessary there. Removed.

**Pull request closed without merging**